resources/ext/config/conf.d/puppetserver.conf

```diff
@@ -2,7 +2,7 @@
 jruby-puppet: {
     # Where the puppet-agent dependency places puppet, facter, etc...
     # Puppet server expects to load Puppet from this location
-    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby]
+    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby, /opt/puppetlabs/puppet/cache/lib]
```

Comment:
@cprice404 @camlow325 given we're at puppet 4.0 and server 2.0, wouldn't now be the time to document the breaking change that we already implemented in SERVER-357? The master and agent no longer share libdir, so you can't use the agent to distribute hiera backends like this.

I think the technical issue preventing us from moving forward, is as @hlindberg mentioned, hiera loads its backends using require (since there is no puppet autoloader in that context). I think the way to fix that is for hiera to always use an absolute path to load the backend (like we do when loading helper code from a type, provider, report processor, etc, e.g.

```
require Pathname.new(__FILE__).dirname + "relative/path/to/backend"
```

Then we could just use r10k to distribute the module to the server.